src/commonMain/kotlin/fr/acinq/lightning/channel/states/Normal.kt

```diff
@@ -345,9 +359,99 @@ data class Normal(
                             }
                         }
                     }
+
+                    is Stfu -> if (commitments.remoteIsQuiescent()) {
+                        when (spliceStatus) {
+                            is SpliceStatus.None -> {
+                                if (commitments.localIsQuiescent()) {
+                                    Pair(this@Normal.copy(spliceStatus = SpliceStatus.NonInitiatorQuiescent), listOf(ChannelAction.Message.Send(Stfu(channelId, initiator = false))))
+                                } else {
+                                    Pair(this@Normal.copy(spliceStatus = SpliceStatus.ReceivedStfu(cmd.message)), emptyList())
+                                }
+                            }
+                            is SpliceStatus.QuiescenceRequested -> {
+                                // We could keep track of our splice attempt and merge it with the remote splice instead of cancelling it.
+                                // But this is an edge case that should rarely occur, so it's probably not worth the additional complexity.
+                                logger.warning { "our peer initiated quiescence before us, cancelling our splice attempt" }
+                                spliceStatus.command.replyTo.complete(ChannelCommand.Commitment.Splice.Response.Failure.ConcurrentRemoteSplice)
+                                Pair(this@Normal.copy(spliceStatus = SpliceStatus.ReceivedStfu(cmd.message)), emptyList())
+                            }
+                            is SpliceStatus.InitiatorQuiescent -> {
+                                // if both sides send stfu at the same time, the quiescence initiator is the channel initiator
+                                if (!cmd.message.initiator || commitments.params.localParams.isInitiator) {
+                                    if (commitments.isQuiescent()) {
+                                        val parentCommitment = commitments.active.first()
+                                        val fundingContribution = FundingContributions.computeSpliceContribution(
+                                            isInitiator = true,
+                                            commitment = parentCommitment,
+                                            walletInputs = spliceStatus.command.spliceIn?.walletInputs ?: emptyList(),
+                                            localOutputs = spliceStatus.command.spliceOutputs,
+                                            targetFeerate = spliceStatus.command.feerate
+                                        )
+                                        val commitTxFees = Transactions.commitTxFee(commitments.params.remoteParams.dustLimit, parentCommitment.remoteCommit.spec)
```

Comment:
Actually that is incorrect, we should set this to 0 if we're not the channel initiator (and thus don't pay the commit tx fee). This is a (small) bug on eclair, where we should only take this into account when we're the channel initiator: can you open a PR to fix this on eclair?

Reply:
fixed in [982d5d2](https://github.com/ACINQ/lightning-kmp/pull/558/commits/982d5d22fa484ef7be1bb7925664d50f32a4ca4b) and I'll also open a PR in scala with the same fix.